app/Services/PostService.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+namespace App\Services;
+
+use Illuminate\Support\Facades\Auth;
+use App\Contracts\Dao\PostDaoInterface;
+use App\Contracts\Services\PostServiceInterface;
+
+class PostService implements PostServiceInterface
+{
+    private $postDao;
+    /**
+     * constructor class
+     *
+     * @param PostDaoInterface $postDao
+     */
+    public function __construct(PostDaoInterface $postDao)
+    {
+        $this->postDao = $postDao;
+    }
+
+    /**
+     * Get user list
+     * @return object
+     */
+    public function getPost(): object
+    {
+        if (Auth::guest()) {
+            return $this->postDao->getAllPost();
+        } else {
+            return $this->postDao->getPublicPost();
+        }
+    }
+
+    /**
+     * Get My Post
+     *
+     * @return object
+     */
+    public function getMyPost(): object
+    {
+        return $this->postDao->getMyPost(Auth::user()->id);
+    }
+
+    /**
+     * create post
+     *
+     * @return void
+     */
+    public function createPost(array $data): void
+    {
+        $data['created_by'] = auth()->user()->id ?? null;
+        $data['created_at'] = now();
```

Comment:
You don't need to add created_at on create. 
Add` public $timestamps = true; `on model. 
Laravel automatically add created_at and updated_at time on create and update transaction if model use timestamp.